classes/Product.php

```diff
@@ -431,6 +431,12 @@ class ProductCore extends ObjectModel
      */
     protected static $cacheStock = [];
 
+    /** @var bool */
+    protected static $bulkDeletionMode = false;
```

Comment:
Do you think we can provide an explanation about this variable?

Maybe something very simple, like "This flag is used to acknowledge that deletions are going to happen in bulk and expensive operations like index refreshs should be done only once after all deletions are done" with a reference to `beginBulkDeletionMode()` and `beginBulkDeletionMode()` ?

I think this could be useful to help curious developers to understand this 😄 

@lmeyer1 I remember you tried multiple things linked to optimizations, what do you think about it?

Reply:
This PR looks good to me. Indeed, adding an explanation about the variable is a good choice.